### PR TITLE
Refactoring to use known values

### PR DESCRIPTION
**Pull request by kpratyus**
Replaced hard coded values with known values in ExporterFactory, SamplerFactory, SpanProcessorFactory

### CODE REVIEW

File: src/SDK/Trace/ExporterFactory.php [hunk 1]
```diff
@@ -83,24 +84,24 @@ public function fromEnvironment(): ?SpanExporterInterface
...
-            case 'none':
+            case Values::VALUE_NONE:
                 return null;
-            case 'jaeger':
-            case 'zipkin':
-            case 'newrelic':
+            case Values::VALUE_JAEGER:
+            case Values::VALUE_ZIPKIN:
+            case Values::VALUE_NEWRELIC:
             case 'zipkintonewrelic':
```
Comment (brettmc) [inline src/SDK/Trace/ExporterFactory.php:92]: is there a reason this one was left untouched?

**Comment by tidal** [comment]
Thanks @kpratyus !

If you also change the one value, which Brett mentioned, which should be good to merge.

File: src/SDK/Trace/ExporterFactory.php (hunk 1 above)
Comment (kpratyus) [reply src/SDK/Trace/ExporterFactory.php:92]: The refactoring has been done for those variable which are already defined in the knownvalues inteface. 'zipkintonewrelic' is not defined there. The knownvalues interface only has variables which are defined in the OTEL spec -  https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#exporter-selection. That's why it has been ommitted.

Comment (tidal) [reply src/SDK/Trace/ExporterFactory.php:92]: @kpratyus 
Ok, got you.
I did not remember, that I didn't add those values.
In general our "known values" could have more values than the ones stated in the spec.
However the exporter factory is subject to change in this case anyway as we may have to deal
with more vendor specific exporters in general, so we must have a more dynamic way than hard coding
all the options. Also actually users should be able to use thier custom exporters.

@brettmc 
So, I think we should just merge this as is, and take care of tis later (I ill keep in mind)
Do you agree?

**Review by tidal — APPROVED** [review]

File: src/SDK/Trace/ExporterFactory.php (hunk 1 above)
Comment (brettmc) [reply src/SDK/Trace/ExporterFactory.php:92]: Yes, I'm happy to merge as is

**Pull request merged**